osu.Game/Overlays/Chat/DrawableChannel.cs

```diff
@@ -90,6 +105,24 @@ private void newMessagesArrived(IEnumerable<Message> newMessages)
             }
         }
 
+        private void pendingMessageResolved(Message existing, Message updated)
+        {
+            var found = flow.Children.LastOrDefault(c => c.Message == existing);
+            if (found != null)
+            {
+                Trace.Assert(updated.Id.HasValue, "An updated message was returned with no ID.");
+
+                flow.Remove(found);
+                found.Message = updated;
+                flow.Add(found);

```

Comment:
Is it still necessary to remove and re-add like this? As far as I can see this doesn't do any depth update or the like...